app/views/template_invocations/show.html.erb

```diff
@@ -19,6 +19,7 @@ end
 <div id="title_action">
   <div class="btn-toolbar pull-right">
     <%= link_to(_('Back to Job'), job_invocation_path(@template_invocation.job_invocation), :class => 'btn btn-default') %>
+    <%= link_to(_('Rerun'), rerun_job_invocation_path(@template_invocation.job_invocation, :host_ids => [ @host.id ]), :class => 'btn btn-default') if @host and @host.id.in? @template_invocation.job_invocation.failed_host_ids %>
```

Comment:
While ruby supports `and` and `or` for flow control, `&&` and `||` are usually used.